Flake: provisionerd/TestProvisionerd/ConnectErrorClose: panic: close of closed channel

Seen here: https://github.com/coder/coder/actions/runs/12580484454/job/35062490533

```
    t.go:106: 2025-01-02 09:53:27.093 [debu]  provisionerd: dialing coderd
    t.go:106: 2025-01-02 09:53:27.096 [warn]  provisionerd: coderd client failed to dial ...
        error= an error:
                   github.com/coder/coder/v2/provisionerd_test.TestProvisionerd.func3.2
                       C:/a/coder/coder/provisionerd/provisionerd_test.go:75
    t.go:106: 2025-01-02 09:53:27.096 [info]  provisionerd: closing provisionerd
    t.go:106: 2025-01-02 09:53:27.181 [debu]  provisionerd: waiting for goroutines to exit
    t.go:106: 2025-01-02 09:53:27.180 [debu]  provisionerd: dialing coderd
    t.go:106: 2025-01-02 09:53:27.188 [debu]  provisionerd: connect loop exited
panic: close of closed channel

goroutine 213 [running]:
github.com/coder/coder/v2/provisionerd_test.TestProvisionerd.func3.2({0xc000106540?, 0x1?})
	C:/a/coder/coder/provisionerd/provisionerd_test.go:75 +0xc7
github.com/coder/coder/v2/provisionerd.(*Server).connect(0xc0002f3a20)
	C:/a/coder/coder/provisionerd/provisionerd.go:225 +0x275
created by github.com/coder/coder/v2/provisionerd.New in goroutine 38
	C:/a/coder/coder/provisionerd/provisionerd.go:109 +0x4c5

```

--- a/provisionerd/provisionerd_test.go
+++ b/provisionerd/provisionerd_test.go
@@ -70,8 +70,11 @@ func TestProvisionerd(t *testing.T) {
 			close(done)
 		})
 		completeChan := make(chan struct{})
+		var completed sync.Once
 		closer := createProvisionerd(t, func(ctx context.Context) (proto.DRPCProvisionerDaemonClient, error) {
-			defer close(completeChan)
+			completed.Do(func() {
+				defer close(completeChan)
+			})
 			return nil, xerrors.New("an error")
 		}, provisionerd.LocalProvisioners{})
 		require.Condition(t, closedWithin(completeChan, testutil.WaitShort))

